guides/common/modules/proc_installing-and-configuring-puppet-agent-during-host-registration.adoc

```diff
@@ -0,0 +1,63 @@
+* You created a Product and repository for the upstream Puppet agent, such as `\https://yum.puppet.com` or `\https://apt.puppet.com`, and synchronized the repository to {Project}.
+For more information, see {ContentManagementDocURL}Importing_Content_content-management[Importing Content] in _{ContentManagementDocTitle}_.
+* You created an activation key that enables the Puppet agent repository for hosts.
+For more information, see {ContentManagementDocURL}Managing_Activation_Keys_content-management[Managing Activation Keys] in _{ContentManagementDocTitle}_.
+endif::[]
+
+.Procedure
+. In the {ProjectWebUI}, navigate to *Configure* > *Global Parameters* to add host parameters globally.
+Alternatively, you can navigate to *Configure* > *Host Groups* and edit or create a host group to add host parameters only in a host group.
+ifdef::foreman-el,foreman-deb[]
+. Enable the Puppet agent from the official Puppet repository using one of the following host parameters in global parameters or a host group:
+
+* Add a host parameter named `enable-puppetlabs-repo` for the latest stable Puppet release in the unversioned repository.
+* Add a host parameter named `enable-official-puppet7-repo` for the Puppet 7 release repository.
+* Add a host parameter named `enable-puppetlabs-puppet6-repo` for the Puppet 6 release repository.
+
++
+Select the *boolean* type and set the value to `true`.
+endif::[]
+ifdef::katello,orcharhino,satellite[]
+. Enable the Puppet agent using a host parameter in global parameters or a host group.
+endif::[]
+ifdef::katello,orcharhino[]
+Add a host parameter named `enable-puppet7` for Puppet 7 or `enable-puppet6` for Puppet 6, select the *boolean* type, and set the value to `true`.
+endif::[]
+ifdef::satellite[]
+Add a host parameter named `enable-puppet7` for Puppet 7, select the *boolean* type, and set the value to `true`.
+endif::[]
+. Specify configuration for the Puppet agent using all the following host parameters in global parameters or a host group:
+* Add a host parameter named `puppet_server`, select the *string* type, and set the value to the hostname of your {ProjectServer} or any {SmartProxyServer} that you use as the Puppet server, such as `{foreman-example-com}`.
+You can also use this parameter to configure the Puppet agent for a non-integrated Puppet server.
+* Add a host parameter named `puppet_ca_server`, select the *string* type, and set the value to the hostname of your {ProjectServer} or any {SmartProxyServer} that you use as the Puppet CA server, such as `{foreman-example-com}`.
+You can also use this parameter to configure the Puppet agent for a non-integrated Puppet CA server.
+* Add a host parameter named `puppet_environment`, select the *string* type, and set the value to the Puppet environment you want the host to use.
+ifdef::katello,orcharhino,satellite[]
+. Navigate to *Hosts* > *Register Host* and register your host using an appropriate activation key.
+endif::[]
+ifdef::foreman-el,foreman-deb[]
```

Comment:
When it's a matching section (as is this case), I'd recommend making it a negation of it. More concrete: make this the `ifndef` matching the `ifdef` on line 49
```suggestion
ifndef::katello,orcharhino,satellite[]
```

Essentially turning it into an if/else statement, rather than two if statements.